mne/preprocessing/xdawn.py

```diff
@@ -62,6 +69,12 @@ def _least_square_evoked(epochs_data, events, tmin, sfreq):
         An concatenated array of toeplitz matrix for each event type.
     """
 
+    if isinstance(solver, string_types):
```

Comment:
IMO this check is quite unnecessary. This is a private function so you can just check
```
if solver == "pinv":
    ...
elif solver == 'cholesky':
    ...
```